js/ssai_pulse.js

```diff
@@ -433,6 +438,20 @@ OO.Ads.manager(function(_, $)
       if (_.has(adIdVastData, id3Object.adId))
       {
         var adObject = adIdVastData[id3Object.adId];
+
+        // If the id3object duration was a bad value, reapply the timeout to the new
+        // duration
+        var duration = _normalizeDuration(id3Object, adObject);
+        if (duration !== id3Object.duration)
```

Comment:
Can we add a check that the duration is valid (is a number, plus whatever validations are needed)?